apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -116,6 +116,8 @@
 
 -callback open_db(db(), create_db_opts()) -> ok | {error, _}.
 
+-callback update_db_config(db(), create_db_opts()) -> ok | {error, _}.
```

Comment:
The "update" naming hides any other applications of generation usage, like retention; do we intend this?

@lafirest  @ieQu1 what do you think?

Reply:
Sorry, I'm not sure I understood the question. This callback signature looks fine to me.

Reply:
In other words, do we plan to create new generations _only_ when we want to update the config?

Reply:
I see, you're right. We may indeed need to create new generations without updating the config.